content/posts/2022-11-09-iOS16-lock-screen-dynamic-time/index.md

```diff
@@ -0,0 +1,185 @@
+import dayjs from 'dayjs';
+```
+
+- **Display current date**
+  - To display the date, we can use `dayjs().format("dddd, DD MMMM")` that will format the date like `Friday, 05 November`
+  - To display the time, we can use `dayjs().format("hh:mm")` that will format the time like `12:00`
+
+With these changes, the texts at the top of our component should be,
+
+```html
+<Text style="{styles.date}">{dayjs().format("dddd, DD MMMM")}</Text>
+<Text style="{styles.time}">{dayjs().format("hh:mm")}</Text>
+```
+
+Cool! Now we should be able to see the current date and time in our simulators or devices. However, the work is not done yet. As you have probably discovered that our date-time is not really updating, even though it gives us the current date-time when the application renders but after that it does not update anymore.
+
+Let’s now fix it. We don’t want a broken clock in our application. Do we?
+
+## Update the date-time
+
+We need a couple of changes now in order to update the time.
+
+### Create a date-time state
+
+1. Since we will need to update our time, we need to put it into a state variable.
+
+```jsx
+import { useState } from 'react';
+
+// inside our component
+const [date, setDate] = useState(dayjs()); // default value is now our current time
+```
+
+### Format our date-time using dayjs
+
+2. We have put `dayjs()` into our `date` state so we can use our `date` state to show the date-time now inside our `<Text />` component
+
+```jsx
+<Text style={styles.date}>{date.format("dddd, DD MMMM")}</Text>
+<Text style={styles.time}>{date.format("hh:mm")}</Text>
+
+```
+
+### Update the date-time state using `setInterval`
+
+3. We want to update the time with one minute interval every time. To do so we can use the javascript’s `setInterval` function. A typical `setInterval` function’s prototype is the following,
+
+```jsx
+setInterval(() => {
+  functionToTrigger();
+}, timeInMs);
+```
+
+This means that the `functionToTrigger()` will be called or triggered every `timeInMs` value. We will set the value of `timeInMs` and the `functionToTrigger` now.
+
+As we want to update our clock every minute, our function will become,
+
+```jsx
+setInterval(() => {
+  updateTime(); // dummy function for now, will work on this.
+}, 60000); // 1 min = 60000 milliseconds
+```
+
+Great, now our `updateTime` function will trigger an update in every one minute. Let’s update our time by using `setDate` hook.
+
+```jsx
+setInterval(() => {
+  setDate(dayjs()); // dayjs() always gives us the current time
+}, 60000);
+```
+
+4. We have our core logic set up for updating the time, where do we want to use this logic now? You might have guessed it already, since this is a side effect and it does not depend on user’s interaction we will use react’s `useEffect` hook. The next question is when do we want to start this logic? We want to start our timer after our component mounts. By using an empty dependency array inside our `useEffect` hook we can achieve that lifecycle.
```

Comment:
I am a bit lost about how the counting of the chapters works. Is it how it's supposed to be?

Reply:
Yeah, numbers don't make sense. Just kept it simple now. Removed the numbers